scipy/optimize/tests/test_optimize.py

```diff
@@ -1233,45 +1233,49 @@ def test_attributes_present(self):
                 assert_(attribute in dir(res))
 
 
+def f1(z, *params):
+    x, y = z
+    a, b, c, d, e, f, g, h, i, j, k, l, scale = params
+    return (a * x**2 + b * x * y + c * y**2 + d*x + e*y + f)
+
+
+def f2(z, *params):
+    x, y = z
+    a, b, c, d, e, f, g, h, i, j, k, l, scale = params
+    return (-g*np.exp(-((x-h)**2 + (y-i)**2) / scale))
+
+
+def f3(z, *params):
+    x, y = z
+    a, b, c, d, e, f, g, h, i, j, k, l, scale = params
+    return (-j*np.exp(-((x-k)**2 + (y-l)**2) / scale))
+
+
+def brute_func(z, *params):
+    return f1(z, *params) + f2(z, *params) + f3(z, *params)
```

Comment:
I assume you moved this because otherwise the parallel functionality didn't work (pickling class method)? It would be good to add back a `func` method and test that in `test_brute`, to ensure there are no regressions for optimizing with methods.

Reply:
That's correct. You can't pickle a method of an instance.